File: two_pointers/minimum_positive_sum_subarray.py
class Solution:
    def minimumSumSubarray(self, nums: List[int], l: int, r: int) -> int:
        n = len(nums)
        out = float('inf')
        pref = [0] * (n + 1)
        for i in range(1, n + 1):
            pref[i] = pref[i - 1] + nums[i - 1]

        for left in range(0, n - l + 1):
            for right in range(left + l, min(left + r, n) + 1):
                curSum = pref[right] - pref[left]
                if curSum > 0:
                    out = min(out, curSum)

        return out if out != float('inf') else -1

File: two_pointers/minimum_positive_sum_subarray.rb
def minimum_sum_subarray(nums, l, r)
    n = nums.size
    ans = Float::INFINITY

    (l..r).each do |window_size|
        next if window_size > n

        window_sum = nums[0, window_size].sum
        ans = [ans, window_sum].min if window_sum > 0

        (window_size...n).each do |i|
            window_sum +=  nums[i] - nums[i - window_size]
            ans = [ans, window_sum].min if window_sum > 0
        end
    end

    ans == Float::INFINITY ? -1 : ans
end